### DISCUSSION TITLE
How to align the bounds of a Plotter scene with the bounds of a mesh?

### DISCUSSION THREAD

**Discussion by ghost** [pyvista/pyvista#2400] [Q&A] [answered]
Hi,

Given the `bounds` of a mesh (e.g. here (-0.5, 0.5, -0.5, 0.5)), is there a way to **automatically** set the camera position of the `Plotter` so the `bounds` are aligned with the borders of the screenshot as shown below.

To obtain the screenshot below , I made use of `pl.camera.zoom` but ideally I would like to have it automatically just given `mesh.bounds`

Thanks! 
![image](https://user-images.githubusercontent.com/25748245/160860431-1de7487e-2408-417b-9de8-77a3d85e9efe.png)

![image](https://user-images.githubusercontent.com/25748245/160859911-d1ff188b-ca98-4609-8503-532fa5e32d4b.png)

**Comment by Sriram1023** [comment]
Any updates on this?
Or atleast is there a way to find the appropriate zoom level using object bounds?

**Comment by Sriram1023** [comment, marked as answer]
Found a work around for this. you can use 
plotter.camera.tight() for this.
Please note doing this will change the view to XY plane. So if that is not what you want, then rotate the mesh(es) accordingly around the origin and do tightening.

As for why it reverts to XY plane, i dont know.

**Reply by ghost** [reply]
That's exactly what I was looking for, thanks a lot!

**Comment by MatthewFlamm** [comment]
#3391 adds functionality to use any Cartesian-aligned view for the `Plotter.camera.tight` method.

@Sriram1023 and @Louis-Verret  If you are interested you can test the functionality in that PR. Let me know if you need help in installing from a Github PR branch.
